project-koku/koku
# COST-807: Create the GCP cost report api.

Jira Issue: https://issues.redhat.com/browse/COST-807

This pr is the groundwork foundation for the gcp api. I decided to start with the cost api because it will be the easiest. The group by & filter options are: `["account", "region", "service", "project"]`

## How to test

The first step to testing the cost api endpoint is to create a real gcp source:
- https://github.com/project-koku/koku#run-gcp-scenario

After the source is created, the database will be populated with most up to date information for the account in your .env file.

After the report is downloaded, processed, and the summarization of the data has been complete, we can now check out our new endpoint:

#### One: Test the single group_by with wildcard & specific value
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*) 
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[project]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[region]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[region]=*)

Note: After you test with the wildcard option, choose a value from your cost list and test the single group by options.
Example:
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A)

#### Two: Test filter combinations with wildcard & specific value
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[account]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[account]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[project]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[service]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[service]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[region]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[region]=*)

Note: After you test with the wildcard option, choose a value from your cost list and test filter on a specific value.

#### Three: Test group_by & filter combinations
Example:
- http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*&filter[service]=FBF2-FC68-171A

#### Four: Multiple group_by
Example:
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A&group_by[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A&group_by[project]=*)

#### Five: Test csv export
